core/src/test/java/mediumtest/issues/ShowIssueMediumTests.java

```diff
@@ -0,0 +1,184 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package mediumtest.issues;
+
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import mediumtest.fixtures.ServerFixture;
+import mediumtest.fixtures.SonarLintTestBackend;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.sonarsource.sonarlint.core.clientapi.common.TextRangeDto;
+import org.sonarsource.sonarlint.core.commons.TextRange;
+import org.sonarsource.sonarlint.core.embedded.server.ShowIssueRequestHandler;
+
+import static mediumtest.fixtures.ServerFixture.newSonarQubeServer;
+import static mediumtest.fixtures.SonarLintBackendFixture.newBackend;
+import static mediumtest.fixtures.SonarLintBackendFixture.newFakeClient;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+class ShowIssueMediumTests {
+
+  private static final String ISSUE_KEY = "myIssueKey";
+  private static final String PROJECT_KEY = "projectKey";
+  private static final String CONNECTION_ID = "connectionId";
+  private static final String CONFIG_SCOPE_ID = "configScopeId";
+  public static final String RULE_KEY = "ruleKey";
+  private ServerFixture.Server serverWithIssue = newSonarQubeServer("10.2")
+    .withProject(PROJECT_KEY,
+      project -> project.withBranch("branchName",
+        branch -> branch.withIssue(ISSUE_KEY, RULE_KEY, "msg", "author", "file/path", "OPEN", "", "2023-05-13T17:55:39+0202",
+          new TextRange(1, 0, 3, 4))))
+    .withSourceFile("projectKey:file/path", sourceFile -> sourceFile.withCode("source\ncode\nfile"))
+    .start();
+  private SonarLintTestBackend backend;
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException {
+    backend.shutdown().get();
+    if (serverWithIssue != null) {
+      serverWithIssue.shutdown();
+      serverWithIssue = null;
+    }
+  }
+
+  @Test
+  void it_should_update_the_telemetry_on_show_issue() {
+    var fakeClient = newFakeClient().build();
+    backend = newBackend()
+      .withSonarQubeConnection(CONNECTION_ID, serverWithIssue)
+      .withBoundConfigScope(CONFIG_SCOPE_ID, CONNECTION_ID, PROJECT_KEY)
+      .withEmbeddedServer()
+      .build(fakeClient);
+
+    assertThat(backend.telemetryFilePath())
+      .content().asBase64Decoded().asString()
+      .contains("\"showIssueRequestsCount\":0");
+
+    var showIssueRequestHandler = new ShowIssueRequestHandler(newFakeClient().build(), backend.getConnectionConfigurationRepository(),
+      backend.getConfigurationServiceImpl(), backend.getBindingSuggestionProviderImpl(), backend.getServerApiProvider(),
+      backend.getTelemetryServiceImpl());
+
+    showIssueRequestHandler.showIssue(new ShowIssueRequestHandler.ShowIssueQuery(serverWithIssue.baseUrl(), "projectKey", "issueKey"));
```

Comment:
The idea of a medium test is that you exercise the code from the interfaces with the outside world, so in this case I would be more in favor of really making an HTTP request to the server. The reason is that with medium tests we don't want to rely on "internal classes" to not slow down refactoring if needed